frontend/src/lib/components/Alerts/views/EditAlertModal.tsx

```diff
@@ -167,14 +173,18 @@ export function EditAlertModal({
                                                 <LemonSelect
                                                     fullWidth
                                                     data-attr="alertForm-series-index"
-                                                    options={alertSeries?.map(({ event }, index) => ({
-                                                        label: isBreakdownValid
-                                                            ? 'any breakdown value'
-                                                            : formula
-                                                            ? `Formula (${formula})`
-                                                            : `${alphabet[index]} - ${event}`,
-                                                        value: isBreakdownValid || formula ? 0 : index,
-                                                    }))}
+                                                    options={alertSeries?.map(
+                                                        ({ custom_name, name, event }, index) => ({
+                                                            label: isBreakdownValid
+                                                                ? 'any breakdown value'
+                                                                : formula
+                                                                ? `Formula (${formula})`
+                                                                : `${alphabet[index]} - ${
+                                                                      custom_name ?? name ?? event
```

Comment:
You can use `getDisplayNameFromEntityNode` to get a formatted name that uses readable names, such as "Pageview" instead of "$pageview".